internal/retry/retry.go

```diff
@@ -181,11 +181,20 @@ func (r *Retryer) runRetryLoop(
 			continue
 		}
 
+		isPreV50CollectionUUIDNotSupportErr := func(err error) bool {
+			return util.IsFailedToParseError(err) && util.HasServerErrorMessage(err, "collectionUUID")
+		}
+		isV50CollectionUUIDNotSupportErr := func(err error) bool {
+			return util.HasServerErrorMessage(err, "collectionUUID is not supported")
+		}
```

Comment:
Why define these as runtime anonymous subs rather than normal functions? Then they could be tested directly.

Reply:
I don't think these functions worth testing and now they are not called anywhere else. We can move them to normal functions if they are shared later.

Reply:
FWIW, I respectfully disagree that they should be defined inline unless they need to be called elsewhere.

Making these unexported, named functions will minimize bloat in this function and prevent inadvertent closures around higher-scoped variables.

> I don't think these functions worth testing

Again, I’d put the burden of proof the other way: assuming it’s quick & easy to write unit tests for these, why not?

Reply:
Closing this thread since it’s probably not worth blocking.